docs/core_docs/docs/tutorials/chatbot.ipynb

```diff
@@ -1,1149 +1,1448 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [
     {
-      "cell_type": "markdown",
-      "metadata": {},
-      "source": [
-        "If you take a look at LangSmith, you can see exactly what is happening under the hood in the [LangSmith trace](https://smith.langchain.com/public/bf1b1a10-0fe0-42f6-9f0f-b70d9f7520dc/r)."
-      ]
-    },
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[\n",
+      "  SystemMessage {\n",
+      "    \"content\": \"you're a good assistant\",\n",
+      "    \"additional_kwargs\": {},\n",
+      "    \"response_metadata\": {}\n",
+      "  },\n",
+      "  HumanMessage {\n",
+      "    \"content\": \"I like vanilla ice cream\",\n",
+      "    \"additional_kwargs\": {},\n",
+      "    \"response_metadata\": {}\n",
+      "  },\n",
+      "  AIMessage {\n",
+      "    \"content\": \"nice\",\n",
+      "    \"additional_kwargs\": {},\n",
+      "    \"response_metadata\": {},\n",
+      "    \"tool_calls\": [],\n",
+      "    \"invalid_tool_calls\": []\n",
+      "  },\n",
+      "  HumanMessage {\n",
+      "    \"content\": \"whats 2 + 2\",\n",
+      "    \"additional_kwargs\": {},\n",
+      "    \"response_metadata\": {}\n",
+      "  },\n",
+      "  AIMessage {\n",
+      "    \"content\": \"4\",\n",
+      "    \"additional_kwargs\": {},\n",
+      "    \"response_metadata\": {},\n",
+      "    \"tool_calls\": [],\n",
+      "    \"invalid_tool_calls\": []\n",
+      "  },\n",
+      "  HumanMessage {\n",
+      "    \"content\": \"thanks\",\n",
+      "    \"additional_kwargs\": {},\n",
+      "    \"response_metadata\": {}\n",
+      "  },\n",
+      "  AIMessage {\n",
+      "    \"content\": \"no problem!\",\n",
+      "    \"additional_kwargs\": {},\n",
+      "    \"response_metadata\": {},\n",
+      "    \"tool_calls\": [],\n",
+      "    \"invalid_tool_calls\": []\n",
+      "  },\n",
+      "  HumanMessage {\n",
+      "    \"content\": \"having fun?\",\n",
+      "    \"additional_kwargs\": {},\n",
+      "    \"response_metadata\": {}\n",
+      "  },\n",
+      "  AIMessage {\n",
+      "    \"content\": \"yes!\",\n",
+      "    \"additional_kwargs\": {},\n",
+      "    \"response_metadata\": {},\n",
+      "    \"tool_calls\": [],\n",
+      "    \"invalid_tool_calls\": []\n",
+      "  }\n",
+      "]\n"
+     ]
+    }
+   ],
+   "source": [
+    "import { SystemMessage, HumanMessage, AIMessage, trimMessages } from \"@langchain/core/messages\"\n",
+    "\n",
+    "const trimmer = trimMessages({\n",
+    "  maxTokens: 10,\n",
+    "  strategy: \"last\",\n",
+    "  tokenCounter: (msgs) => msgs.length,\n",
+    "  includeSystem: true,\n",
+    "  allowPartial: false,\n",
+    "  startOn: \"human\",\n",
+    "})\n",
+    "\n",
+    "const messages = [\n",
+    "    new SystemMessage(\"you're a good assistant\"),\n",
```

Comment:
Should we just make these OpenAI dicts to remove the need to import?

Reply:
Believe `trimmer` currently requires the message objects